From 24864e37dc0631e6db0b1df181515520d27c3b2b Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Wed, 23 Sep 2020 16:41:43 +0200
Subject: [PATCH] Don't hardcode the CoreFoundation path

This would otherwise hardcode the Xcode version and we would have to recompile after each Xcode update.

Fixes #510
---
 CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6e7b221..434261e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -92,7 +92,9 @@ else ()
             message(FATAL_ERROR "Core Foundation not found")
         endif ()
 
-        list(APPEND PLATFORM_LIBS dl Threads::Threads ${CORE_FOUNDATION_LIB})
+        list(APPEND PLATFORM_LIBS dl Threads::Threads)
+        # Don't add the exact path to CoreFoundation as this would hardcode the SDK version
+        list(APPEND PLATFORM_LINK_FLAGS -framework CoreFoundation)
     elseif (${CMAKE_SYSTEM_NAME} STREQUAL "Linux") # Android does not link to libpthread nor librt, so this is fine
         list(APPEND PLATFORM_LIBS dl m Threads::Threads rt)
     elseif(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
@@ -155,6 +157,7 @@ add_library(${PROJECT_NAME} ${COMMON_SRC})
 aws_set_common_properties(${PROJECT_NAME} NO_WEXTRA)
 aws_prepare_symbol_visibility_args(${PROJECT_NAME} "AWS_COMMON")
 target_compile_options(${PROJECT_NAME} PUBLIC ${PLATFORM_CFLAGS})
+target_link_options(${PROJECT_NAME} PUBLIC ${PLATFORM_LINK_FLAGS})
 
 aws_check_headers(${PROJECT_NAME} ${AWS_COMMON_HEADERS} ${AWS_TEST_HEADERS} ${AWS_COMMON_OS_HEADERS})
 
-- 
2.29.2

